===================================================================
RCS file: /sources/findutils/findutils/find/ftsfind.c,v
retrieving revision 1.23
retrieving revision 1.24
diff -u -r1.23 -r1.24
--- findutils/find/ftsfind.c	2007/04/15 18:57:40	1.23
+++ findutils/find/ftsfind.c	2007/04/17 08:42:31	1.24
@@ -445,8 +445,9 @@
 	}
     }
   
-  /* Not an error, cope with the usual cases. */
-  if (ent->fts_info == FTS_NSOK)
+  /* Cope with the usual cases. */
+  if (ent->fts_info == FTS_NSOK
+      || ent->fts_info == FTS_NS /* e.g. symlink loop */)
     {
       assert(!state.have_stat);
       assert(!state.have_type);
@@ -717,6 +718,6 @@
 is_fts_enabled(int *fts_options)
 {
   /* this version of find (i.e. this main()) uses fts. */
-  *fts_options = fts_options;
+  *fts_options = ftsoptions;
   return true;
 }
